### PR TITLE
rustc: replace `TyCtxt<'a, 'gcx, 'tcx>` with `TyCtxt<'gcx, 'tcx>`.

### PR DESCRIPTION
**Pull request by eddyb**
This first lifetime parameter of `TyCtxt` has been phantom for a while, thanks to @Zoxc, but was never removed, and I'm doing this now in preparation for removing the `'gcx`/`'tcx` split.

I wasn't going to do this as a separate step, and instead start converting uses of `TyCtxt` to a single-lifetime alias of it (e.g. `type TyCx<'tcx> = TyCtxt<'tcx, 'tcx, 'tcx>;`) but it turns out (as @Zoxc rightly predicted) that there is far more fallout from not needing a lifetime for the first parameter of `TyCtxt`.

That is, going from `TyCtxt<'a, 'gcx, 'tcx>` to `TyCtxt<'tcx, 'gcx, 'tcx>` (the first commit in this PR) has the largest amount of fallout out of all the changes we might make (because it can require removing the `'a` parameter of `struct`s containing `tcx: TyCtxt<'a, ...>`), and is the hardest to automate (because `'a` is used everywhere, not just with `TyCtxt`, unlike, say `'gcx, 'tcx` -> `'tcx`).

So I'm submitting this now to get it out of the way and reduce further friction in the future.

**EDIT**: for the `rustfmt` commit, I used https://github.com/rust-lang/rustfmt/issues/1324#issuecomment-482109952, and manually filtered out some noise, like in #61735, but unlike that PR, there was also a weird bug to work around.
It should be reviewed separately, and dropped if unwanted.

cc @rust-lang/compiler r? @nikomatsakis

### CODE REVIEW

**Comment by eddyb** [comment]
This might error because of `rustfmt`, not sure why it introduced syntax errors, I'll dig into it tomorrow.

**Comment by estebank** [comment]
@eddyb there are a few of these: `visitor.cx = prev_}`

https://github.com/rust-lang/rust/pull/61722/commits/0472b6c3157cdb0714df3fbc08945e3106679943#diff-eaec58cd2e0c9e0564f2f3f7e417340cR818
https://github.com/rust-lang/rust/pull/61722/commits/0472b6c3157cdb0714df3fbc08945e3106679943#diff-eaec58cd2e0c9e0564f2f3f7e417340cR839
https://github.com/rust-lang/rust/pull/61722/commits/0472b6c3157cdb0714df3fbc08945e3106679943#diff-eaec58cd2e0c9e0564f2f3f7e417340cR855
https://github.com/rust-lang/rust/pull/61722/commits/0472b6c3157cdb0714df3fbc08945e3106679943#diff-eaec58cd2e0c9e0564f2f3f7e417340cR873
https://github.com/rust-lang/rust/pull/61722/commits/0472b6c3157cdb0714df3fbc08945e3106679943#diff-eaec58cd2e0c9e0564f2f3f7e417340cR975

I don't know what `rustfix` was trying to do there :-/

**Comment by eddyb** [comment]
@estebank Maybe passing `--file-lines` makes it pick lines to change in an incorrect way.

File: src/librustc_codegen_llvm/base.rs [hunk 1]
```diff
@@ -42,7 +42,7 @@ use rustc::hir::CodegenFnAttrs;
 use crate::value::Value;
 
 pub fn write_compressed_metadata<'a, 'gcx>(
```
Comment (oli-obk) [inline src/librustc_codegen_llvm/base.rs]: `'a` isn't necessary anymore

File: src/librustc_codegen_llvm/base.rs [hunk 2]
```diff
@@ -124,7 +124,7 @@ pub fn compile_codegen_unit(tcx: TyCtxt<'a, 'tcx, 'tcx>, cgu_name: InternedStrin
     submit_codegened_module_to_llvm(&LlvmCodegenBackend(()), tcx, module, cost);
 
     fn module_codegen<'ll, 'tcx>(
```
Comment (oli-obk) [inline src/librustc_codegen_llvm/base.rs]: `'ll` isn't necessary anymore

**Review by oli-obk — COMMENTED** [review]
r=me with leftover unnecessary lifetimes removed

File: src/librustc_codegen_llvm/base.rs (hunk 1 above)
Comment (eddyb) [reply src/librustc_codegen_llvm/base.rs]: That's weird, why wouldn't it get an error?

Comment (eddyb) [reply src/librustc_codegen_llvm/base.rs]: Oh, the codegen crates are missing `#![deny(internal)]` and `#![deny(unused_lifetimes)]`...

File: src/librustc/infer/nll_relate/mod.rs [hunk 3]
```diff
@@ -494,11 +494,11 @@ impl VidValuePair<'tcx> for (Ty<'tcx>, ty::TyVid) {
     }
 }
 
-impl<D> TypeRelation<'me, 'gcx, 'tcx> for TypeRelating<'me, 'gcx, 'tcx, D>
+impl<D> TypeRelation<'gcx, 'tcx> for TypeRelating<'me, 'gcx, 'tcx, D>
```
[the remark this answers is not in the document]
Comment (eddyb) [reply src/librustc/infer/nll_relate/mod.rs]: I believe that'd be handled by the "single-use lifetime" lint, which we haven't turned on yet.

**Review by oli-obk — APPROVED** [review]

**Pull request merged**